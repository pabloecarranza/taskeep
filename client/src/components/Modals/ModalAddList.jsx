import React from 'react';
import {
	Modal,
	ModalContent,
	ModalHeader,
	ModalFooter,
	ModalBody,
	ModalCloseButton,
	Button,
	FormControl,
	FormLabel,
	Input,
	useToast,
} from '@chakra-ui/react';
import { usePostListMutation } from '../../features/api/listSlice';

export const ModalAddList = ({ isOpen, onClose }) => {
	const toast = useToast();
	const handleClick = () => {
		onClose();
	};

	const initialRef = React.useRef(null);
	const [input, setInput] = React.useState({
		name: '',
	});

	const handleChange = e => {
		setInput({ name: e.target.value });
	};

	const [PostList, PostListResponse] = usePostListMutation();

	const handleSubmit = async () => {
		console.log(input.name);
		if (input.name.length === 0) {
			toast({
				title: 'Error.',
				description: 'The field list name cant by empty.',
				status: 'error',
				duration: 2000,
				isClosable: true,
			});

			return;
		} else {
			PostList(input)
				.unwrap()
				.then(respon => {
					toast({
						title: 'Success.',
						description: `${respon.message}`,
						status: 'success',
						duration: 3000,
						isClosable: true,
					});
					onClose();
				})
				.catch(error => {
					toast({
						title: 'Error',
						description: `${error.data.message}`,
						status: 'error',
						duration: 2000,
						isClosable: true,
					});
				});

			/* 			
			SignIn(credentials)
				.unwrap()
				.then(respon => {
					toast({
						title: 'Success.',
						description: `${respon.message}`,
						status: 'success',
						duration: 2000,
						isClosable: true,
					});
				})
				.catch(error => {
					toast({
						title: 'Error',
						description: `${error.data.message}`,
						status: 'error',
						duration: 2000,
						isClosable: true,
					});
				});
			setCredentials({
				username: '',
				password: '',
				email: '',
			});
			*/
			setInput({ name: '' });
		}
	};

	return (
		<>
			<Modal
				initialFocusRef={initialRef}
				isOpen={isOpen}
				onClose={handleClick}
				isCentered
			>
				<ModalContent bg='#1A202C'>
					<ModalHeader>Create a new task list</ModalHeader>
					<ModalCloseButton />
					<ModalBody pb={6}>
						<FormControl>
							<FormLabel>List name</FormLabel>
							<Input
								ref={initialRef}
								placeholder='Write here...'
								value={input.name}
								onChange={handleChange}
							/>
						</FormControl>
					</ModalBody>

					<ModalFooter>
						<Button variant='ghost' onClick={onClose}>
							Cancel
						</Button>
						<Button colorScheme='blue' mr={3} onClick={handleSubmit} ml='4px'>
							Save
						</Button>
					</ModalFooter>
				</ModalContent>
			</Modal>
		</>
	);
};
