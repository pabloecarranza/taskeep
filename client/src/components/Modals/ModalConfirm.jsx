import React from 'react';
import {
	Modal,
	ModalContent,
	ModalHeader,
	ModalFooter,
	ModalBody,
	ModalCloseButton,
	Button,
} from '@chakra-ui/react';
import useDeleteList from './../../Hooks/useDeleteList';
import PropTypes from 'prop-types';

export const ModalConfirm = ({
	isOpen,
	onClose,
	onOpen,
	List,
	setSeleted,
	title_partOne,
	title_partTwo,
	message,
	button_one,
	button_two,
}) => {
	const initialRef = React.useRef(null);

	const [deleteList] = useDeleteList(onClose, setSeleted);

	return (
		<>
			<Modal
				initialFocusRef={initialRef}
				isOpen={isOpen}
				onClose={onClose}
				onOpen={onOpen}
				isCentered
			>
				<ModalContent bg='#1A202C'>
					<ModalHeader>{title_partOne + List.name + title_partTwo}</ModalHeader>
					<ModalCloseButton />
					<ModalBody pb={6}>{message}</ModalBody>

					<ModalFooter>
						<Button variant='ghost' onClick={onClose}>
							{button_one}
						</Button>
						<Button
							colorScheme='red'
							mr={3}
							onClick={() => deleteList(List)}
							ml='4px'
						>
							{button_two}
						</Button>
					</ModalFooter>
				</ModalContent>
			</Modal>
		</>
	);
};

ModalConfirm.propTypes = {
	onClose: PropTypes.func.isRequired,
	onOpen: PropTypes.func.isRequired,
	List: PropTypes.object.isRequired,
	setSeleted: PropTypes.func.isRequired,
	title_partOne: PropTypes.string.isRequired,
	title_partTwo: PropTypes.string.isRequired,
	message: PropTypes.string.isRequired,
	button_one: PropTypes.string.isRequired,
	button_two: PropTypes.string.isRequired,
};
