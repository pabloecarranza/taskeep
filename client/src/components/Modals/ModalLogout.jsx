import React from 'react';
import {
	Modal,
	ModalContent,
	ModalHeader,
	ModalFooter,
	ModalBody,
	ModalCloseButton,
	Button,
	ModalOverlay,
} from '@chakra-ui/react';
import { sessionOut } from '../../features/api/sessionSlice';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

export const ModalLogout = ({
	isOpen,
	onClose,
	onOpen,
	title,
	message,
	button_one,
	button_two,
}) => {
	const initialRef = React.useRef(null);
	const dispatch = useDispatch();
	let navigate = useNavigate();

	const logoutConfirm = () => {
		dispatch(sessionOut());
		onClose();
		navigate('/', { replace: true });
	};

	return (
		<>
			<Modal
				initialFocusRef={initialRef}
				isOpen={isOpen}
				onClose={onClose}
				onOpen={onOpen}
				isCentered
			>
				<ModalOverlay />
				<ModalContent bg='#1A202C'>
					<ModalHeader>{title}</ModalHeader>
					<ModalCloseButton />
					<ModalBody pb={6}>{message}</ModalBody>

					<ModalFooter>
						<Button variant='ghost' onClick={onClose}>
							{button_one}
						</Button>
						<Button colorScheme='red' mr={3} onClick={logoutConfirm} ml='4px'>
							{button_two}
						</Button>
					</ModalFooter>
				</ModalContent>
			</Modal>
		</>
	);
};

ModalLogout.propTypes = {
	onClose: PropTypes.func.isRequired,
	onOpen: PropTypes.func.isRequired,
	List: PropTypes.object.isRequired,
	title: PropTypes.string.isRequired,
	message: PropTypes.string.isRequired,
	button_one: PropTypes.string.isRequired,
	button_two: PropTypes.string.isRequired,
};
