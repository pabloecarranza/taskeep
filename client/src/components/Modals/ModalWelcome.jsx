import React from 'react';
import {
	Modal,
	ModalOverlay,
	ModalContent,
	ModalHeader,
	ModalFooter,
	ModalBody,
	ModalCloseButton,
	Button,
	Center,
	Image,
} from '@chakra-ui/react';
import me from '../../assets/me.jpg';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

export const ModalWelcome = ({
	isOpen,
	onClose,
	setIsloaded,
	greeting,
	welcome_text_1,
	welcome_text_2,
	name,
	button_text,
}) => {
	const handleClick = () => {
		onClose();
		setIsloaded(v => !v);
	};

	const userData = useSelector(state => state.session);

	return (
		<Modal isOpen={isOpen} onClose={handleClick} isCentered size='lg'>
			<ModalOverlay />
			<ModalContent bg='#1A202C'>
				<ModalHeader>
					{greeting} {userData ? userData.username : ''}!
				</ModalHeader>
				<ModalCloseButton />
				<ModalBody>
					<Center>
						{welcome_text_1}
						<br />
						{welcome_text_2}
						<Image src={me} alt='PabloCarranza' w='30%' />
					</Center>
					<br />
					{name}
				</ModalBody>

				<ModalFooter>
					<Button colorScheme='blue' mr={3} onClick={handleClick}>
						{button_text}
					</Button>
				</ModalFooter>
			</ModalContent>
		</Modal>
	);
};

ModalWelcome.propTypes = {
	greeting: PropTypes.string.isRequired,
	welcome_text_1: PropTypes.string.isRequired,
	welcome_text_2: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	button_text: PropTypes.string.isRequired,
	onClose: PropTypes.func.isRequired,
	setIsloaded: PropTypes.func.isRequired,
	isOpen: PropTypes.bool.isRequired,
};
