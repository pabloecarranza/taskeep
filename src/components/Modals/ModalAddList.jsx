import React from 'react';
import {
	Modal,
	ModalContent,
	ModalHeader,
	ModalFooter,
	ModalBody,
	ModalCloseButton,
	Button,
	FormControl,
	FormLabel,
	Input,
} from '@chakra-ui/react';
import PropTypes from 'prop-types';
import usehandleAdd from '../../Hooks/useHandleAdd';

export const ModalAddList = ({
	isOpen,
	onClose,
	title,
	name,
	buttonOne,
	buttonTwo,
	placeHolder,
}) => {
	const [input, handleChange, handleSubmit, handleClick] =
		usehandleAdd(onClose);

	const initialRef = React.useRef(null);

	return (
		<>
			<Modal
				initialFocusRef={initialRef}
				isOpen={isOpen}
				onClose={handleClick}
				isCentered
			>
				<ModalContent bg='#1A202C'>
					<ModalHeader>{title}</ModalHeader>
					<ModalCloseButton />
					<ModalBody pb={6}>
						<FormControl>
							<FormLabel> {name} </FormLabel>
							<Input
								ref={initialRef}
								placeholder={placeHolder}
								value={input.name}
								onChange={e => handleChange(e.target.value)}
								onKeyDown={e => handleSubmit(e)}
							/>
						</FormControl>
					</ModalBody>

					<ModalFooter>
						<Button variant='ghost' onClick={onClose}>
							{buttonOne}
						</Button>
						<Button
							colorScheme='blue'
							mr={3}
							ml='4px'
							onClick={e => handleSubmit(e)}
							onKeyDown={e => handleSubmit(e)}
						>
							{buttonTwo}
						</Button>
					</ModalFooter>
				</ModalContent>
			</Modal>
		</>
	);
};

ModalAddList.propTypes = {
	isOpen: PropTypes.bool.isRequired,
	onClose: PropTypes.func.isRequired,
	title: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	buttonOne: PropTypes.string.isRequired,
	buttonTwo: PropTypes.string.isRequired,
	placeHolder: PropTypes.string.isRequired,
};
