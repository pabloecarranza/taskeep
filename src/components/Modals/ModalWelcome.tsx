import {
	Modal,
	ModalOverlay,
	ModalContent,
	ModalHeader,
	ModalFooter,
	ModalBody,
	ModalCloseButton,
	Button,
	Center,
	Image,
} from '@chakra-ui/react';
import me from '../../assets/me.jpg';
import PropTypes from 'prop-types';
import { useCustomSelector } from '../../Hooks/reduxHooks';

interface Props {
	greeting: string,
	welcomeTextOne: string,
	welcomeTextTwo:string,
	name: string,
	buttonText: string,
	onClose: () => void,
	setIsloaded: (cb: (value: boolean) => boolean ) => void,
	isOpen: boolean,
}


export const ModalWelcome = ({
	isOpen,
	onClose,
	setIsloaded,
	greeting,
	welcomeTextOne,
	welcomeTextTwo,
	name,
	buttonText,
}: Props) => {
	
	const handleClick = () :void  => {
		onClose();
		setIsloaded(value => !value);
	};

	const { username } = useCustomSelector((state) => state.session);

	return (
		<Modal isOpen={isOpen} onClose={handleClick} isCentered size='lg'>
			<ModalOverlay />
			<ModalContent bg='#1A202C'>
				<ModalHeader>
					{greeting} {username || ''}!
				</ModalHeader>
				<ModalCloseButton />
				<ModalBody>
					<Center>
						{welcomeTextOne}
						<br />
						{welcomeTextTwo}
						<Image src={me} alt='PabloCarranza' w='30%' />
					</Center>
					<br />
					{name}
				</ModalBody>

				<ModalFooter>
					<Button colorScheme='blue' mr={3} onClick={handleClick}>
						{buttonText}
					</Button>
				</ModalFooter>
			</ModalContent>
		</Modal>
	);
};

ModalWelcome.propTypes = {
	greeting: PropTypes.string.isRequired,
	welcomeTextOne: PropTypes.string.isRequired,
	welcomeTextTwo: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	buttonText: PropTypes.string.isRequired,
	onClose: PropTypes.func.isRequired,
	setIsloaded: PropTypes.func.isRequired,
	isOpen: PropTypes.bool.isRequired,
};
