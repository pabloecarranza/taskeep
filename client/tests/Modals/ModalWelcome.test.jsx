import { fireEvent, render, screen } from '@testing-library/react';
import { vi, expect, describe, it } from 'vitest';
import { ModalWelcome } from './../../src/components/Modals/ModalWelcome';

describe('Test suite on ModalWelcome component', () => {
	const ModalWelcomeDates = {
		greeting: 'Hi',
		welcome_text_1: 'Welcome to my personal app for task management.',
		welcome_text_2:
			'The main functionalities are to create tasks and group them by lists, set the priority and expiration date. I hope you enjoy it !',
		name: 'Pablo',
	};

	it('must show the gretting text', () => {
		const { container } = render(<ModalWelcome {...ModalWelcomeDates} />);
		expect(container).toMatchSnapshot();
	});
});
