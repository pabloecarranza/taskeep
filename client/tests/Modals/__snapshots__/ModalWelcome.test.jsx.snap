// Vitest Snapshot v1

exports[`Test suite on ModalWelcome component > must show the gretting text 1`] = `<div />`;
